src/schema/v2/me/sendConfirmationEmailMutation.ts

```diff
@@ -0,0 +1,77 @@
+import { GraphQLObjectType, GraphQLString, GraphQLUnionType } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import {
+  GravityMutationErrorType,
+  formatGravityError,
+} from "lib/gravityErrorHandler"
+
+const SendConfirmationEmailMutationSuccess = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "SendConfirmationEmailMutationSuccess",
+  isTypeOf: data => data.id,
+  fields: () => ({
+    confirmationSentAt: {
+      type: GraphQLString,
+      resolve: ({ confirmation_sent_at }) => confirmation_sent_at,
+    },
+    unconfirmedEmail: {
+      type: GraphQLString,
+      resolve: ({ unconfirmed_email }) => unconfirmed_email,
+    },
+  }),
+})
+
+const SendConfirmationEmailMutationFailure = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "SendConfirmationEmailMutationFailure",
+  isTypeOf: data => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: err => err,
+    },
+  }),
+})
+
+export const sendConfirmationEmailMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "SendConfirmationEmailMutation",
+  description: "Send a confirmation email",
+  inputFields: {}, // TODO: Is this the right way to say "This mutation does not take any arguments"?
+  outputFields: {
+    confirmationOrError: {
+      type: new GraphQLUnionType({
+        name: "SendConfirmationEmailMutationType",
+        types: [
+          SendConfirmationEmailMutationSuccess,
+          SendConfirmationEmailMutationFailure,
+        ],
+      }),
+      resolve: result => result,
+    },
+  },
+  mutateAndGetPayload: (_, context) => {
+    const { confirmationEmailLoader } = context as any
```

Comment:
For some reason I was getting a Type error although [the loader clearly exists](https://github.com/artsy/metaphysics/pull/2379/files#diff-bd81489154ac1708c130d722ee0f988dR60):

```
src/schema/v2/me/sendConfirmationEmailMutation.ts:74:12 - error TS2722: Cannot invoke an object which is possibly 'undefined'.

74     return emailConfirmationLoader()
              ~~~~~~~~~~~~~~~~~~~~~~~
```

Anyone has insights as to why this is happening?

Reply:
Odd, the snippet you posted has it written as `emailConfirmationLoader`, whereas it's called `confirmationEmailLoader` here in the PR. Perhaps the file wasn't saved or there's some bad state?

Reply:
That was actually a bad or old copy & paste and the name was consistent when I ran it. The fix was to add a guard since the loader could be `undefined`, so TS was actually smart enough to catch an unauthorized state.

```ts
if (!sendConfirmationEmailLoader) {
  throw new Error("You need to be signed in to perform this action")
}
```